tools/mat_setup_workspace.bzl

```diff
@@ -0,0 +1,22 @@
+# Copyright Google Inc. All Rights Reserved.
+#
+# Use of this source code is governed by an MIT-style license that can be
+# found in the LICENSE file at https://angular.io/license
+
+"Install angular source dependencies"
+
+load("@build_bazel_rules_nodejs//:defs.bzl", "yarn_install")
+
+def mat_setup_workspace():
+    """This repository rule should be called from your WORKSPACE file.
+
+    It creates some additional Bazel external repositories that are used internally
+    to build angular material
+    """
+    # Use Bazel managed node modules. See more below: 
+    # https://github.com/bazelbuild/rules_nodejs#bazel-managed-vs-self-managed-dependencies
+    yarn_install(
+        name = "matdeps",
```

Comment:
Could this stay named "npm"? I generally liked that the deps across the project reflected that they were coming from npm

Reply:
That's not possible because it would conflict with the downstream workspace that potentially can also use Bazel managed deps (so it should be somewhat unique)

Reply:
Yea. We have to do the same thing in angular/angular right now. Its built with `ngdeps` since `@npm` would conflict with a user's npm deps downstream. After Ivy is released and we can build downstream from npm published bundles then `@npm` can be used since we'll no longer build Angular + Material from source downstream.